packages/common-scripts/src/anyone_can_pay.ts

```diff
@@ -0,0 +1,549 @@
+import {
+  TransactionSkeletonType,
+  Options,
+  parseAddress,
+  minimalCellCapacity,
+  createTransactionFromSkeleton,
+} from "@ckb-lumos/helpers";
+import {
+  values,
+  Address,
+  Script,
+  QueryOptions,
+  CellProvider,
+  CellCollector as CellCollectorType,
+  OutPoint,
+  HexString,
+  WitnessArgs,
+  core,
+  Cell,
+  utils,
+} from "@ckb-lumos/base";
+import { getConfig, Config } from "@ckb-lumos/config-manager";
+import {
+  isAcpScript,
+  addCellDep,
+  SECP_SIGNATURE_PLACEHOLDER,
+  hashWitness,
+  isAcpAddress,
+} from "./helper";
+import { Reader, normalizers } from "ckb-js-toolkit";
+const { ScriptValue } = values;
+import { Set, List } from "immutable";
+const { CKBHasher, ckbHash, readBigUInt128LE } = utils;
+
+export class CellCollector {
+  private cellCollector: CellCollectorType;
+  private config: Config;
+  public readonly fromScript: Script;
+
+  constructor(
+    fromAddress: Address,
+    cellProvider: CellProvider,
+    {
```

Comment:
Can we skip QueryOptions here? Is there any chance that we have to let users customize the query used here?

Reply:
It's for `anyone-can-pay` with `sudt` https://github.com/nervosnetwork/lumos/pull/69/files#diff-2cf727f7322effda3e47a60c2bbcc253R199

If we skipped `QueryOptions`, we should provide a `sudtToken` param, for search `anyone-can-pay` lock script with `sudt` type script.

Reply:
Okay in that sense let's keep QueryOptions here.